src/main/kotlin/io/github/inductiveautomation/kindling/gatewaynetwork/GatewayNetworkViewer.kt

```diff
@@ -0,0 +1,161 @@
+ * navigate to 'server/static' to view the actual files.
+ *
+ * To get a JSON diagram in Ignition 8.1:
+ * Set the 'gateway.routes.status.GanRoutes' logger set to DEBUG in an Ignition gateway to generate diagram JSON while
+ * viewing the gateway network live diagram page.
+ */
+class GatewayNetworkViewer(path: Path) : ToolPanel() {
+    override val icon = GatewayNetworkTool.icon
+
+    private val browserBtn = JButton("View diagram in browser")
+
+    init {
+        name = path.name
+        toolTipText = path.toString()
+        add(browserBtn)
+
+        val textArea = JTextArea()
+        add(FlatScrollPane(textArea), "newline, push, grow, span")
+
+        // Read the provided file and populate the text area with it
+        path.useLines(DefaultEncoding.currentValue) { lines -> parseLines(lines, textArea) }
+
+        browserBtn.addMouseListener(
+            object : MouseAdapter() {
+                override fun mousePressed(e: MouseEvent?) {
+                    // We need to load the contents from the text area and place them in a file
+                    // with the other static files
+                    val theText = textArea.text
+
+                    // Verify that the provided text is valid JSON, as the user is free to copy and paste in text.
+                    try {
+                        JSONObject(theText)
+                    } catch (e: JSONException) {
+                        JOptionPane.showMessageDialog(
+                            null,
+                            "Error: " + e.message,
+                            "JSON parsing error",
+                            JOptionPane.ERROR_MESSAGE,
+                        )
+                        return
+                    }
+                    JSONObject(theText)
+
+                    val tmpDir: String = writeStaticFiles(theText)
+
+                    val tempAddress = "file://$tmpDir/index.html"
+                    val desktop = if (Desktop.isDesktopSupported()) Desktop.getDesktop() else null
+                    if (desktop != null && desktop.isSupported(Desktop.Action.BROWSE)) {
+                        try {
+                            desktop.browse(URI(tempAddress))
+                        } catch (error: Exception) {
+                            JOptionPane.showMessageDialog(
+                                null,
+                                "Error: " + error.message,
+                                "Error opening browser",
+                                JOptionPane.ERROR_MESSAGE,
+                            )
+                        }
+                    }
+                }
+            },
+        )
+
+        // Temp file cleanup when Kindling exits.
+        val shutdownThread = Thread {
+            val tmpDir: File = Path.of(System.getProperty("java.io.tmpdir"), "gateway-network-diagram").toFile()
+            tmpDir.deleteRecursively()
+        }
+
+        Runtime.getRuntime().addShutdownHook(shutdownThread)
+    }
```

Comment:
Explicit close hooks shouldn't be necessary; you can use `Path.toFile().deleteOnExit()` to do the same thing